roles/setup_lvm_vg/action_plugins/validate_storage.py

```diff
@@ -0,0 +1,160 @@
+  Since the lvg ansible module can reduce a VG automatically, raise an
+  error, asking the operator to manually fix the storage configuration
+  before proceeding
+
+  the following situation need to be handled
+
+  1. Initial setup with 2 disks
+
+  .. code:: shell
+
+    metalk8s_lvm_drives_vg_metalk8s: ['/dev/vdb', '/dev/vdc']
+
+  2. Change the disks with removing a previous one
+
+  .. code:: shell
+
+    metalk8s_lvm_drives_vg_metalk8s: ['/dev/vdb', '/dev/vdd']
+
+  3. Exit with the following error
+
+  ::
+
+    that "/dev/vdc" is part of the VG "vg_metalk8s" and would be removed.
+    Please, either add back "/dev/vdc" to "metalk8s_lvm_drives_vg_metalk8s"
+    or resolve manually the conflict and relaunch the playbooks/deploy.yml
+    playbook like this
+
+    .. code::
+
+      ansible-playbook -i {inventory} -t storage playbooks/deploy.yml
+'''
+
+# Note: to add mode checks/validations, simply add a top-level function whose
+# name starts with `check_`. The function will receive a `task_vars` dictionary
+# as defined by Ansible.
+# Within a `check_*` function, use `assert` to validate values found in
+# `task_vars`, or raise `AssertionError` explicitly. A single check can, of
+# course, contain multiple assertions.
+# Alternatively, for checks which can result in multiple errors, a check can
+# return a list of (or yield) error messages.
+
+from ansible.plugins.action import ActionBase
+
+import inspect
+import sys
+
+
+def is_device_present(device, ansible_devices):
+    '''
+    Check that the device passed in parameters exists
+
+    The device can be either a raw device or a partition but nothing else.
+
+    Params:
+        device (string): The name of the device
+            i.e: 'sdb' or 'sdb1'
+        ansible_devices (dict): The dictionary of devices gather by the 'setup'
+            ansible module
+
+    Returns:
+        bool: True if the device is in ansible_device.
+            False otherwise
+    '''
+
+    # if the device is a raw device, check its presence
+    if device in ansible_devices:
+        return True
+
+    # otherwise, check if it's a partition in other devices
+    for ansible_device, device_attr in ansible_devices.items():
+        if device in device_attr['partitions']:
+            return True
+
+    return False
+
+
+def check_devices_presence(hostvars):
+    '''
+    Check that the devices specified in metalk8S_lvm_drives_<vg name>
+    exists on the server
+
+    Params:
+        hostvars (dict): The dictionary 'hostvars' from 'setup' ansible module
+            for a specific host
+
+    Raises:
+        AssertionError: if the device is not present on the host
+    '''
+
+    for lvm_vg in \
+        hostvars.get(
+            'metalk8s_lvm_all_vgs', {}).keys():
+        mk8s_vg_drives_var = 'metalk8s_lvm_drives_' + lvm_vg
+        for device in hostvars.get(mk8s_vg_drives_var):
+            device_name = device.split('/')[-1]
```

Comment:
I would try to strip the first '/dev/', and check that the result contain no '/'.

Reply:
Good point !